### ISSUE TITLE
Custom Exit Code

### ISSUE THREAD

**Issue by HeroCC** [SpigotMC/BungeeCord#1400]
My feature request is for an API or somehow an ability to set the exit code of the java application. Right now it is set as System.exit( 0 ) regardless of how spigot exits. My request is for the following:
If the player stops the server with /stop and there are no SEVERE warnings, it will exit with an exit code of 0.

* If a plugin stops the server with no exit code it will follow the above
* If a plugin stops the server with an exit code it will follow that exit code, regardless of how the server stops.
* If the player stops the server stops the server with an exit code (/stop 121) it will stop with that exit code, regardless of how the server stops.

Some examples of how this can be used are:
* /reload command being aliased to /stop 5, and a script that reloads the server if the exit code is 5.
* A plugin that stops the server at 2:00 AM with the code of 6, and a monitor that reloads the server after it stops if the code is six.

There are many applications for this feature, and it will be fully embraced by the server community if implemented. Thank you!

This is also requested as a feature for the SpigotAPI [here](https://hub.spigotmc.org/jira/browse/SPIGOT-699).

**Comment by yawkat** [comment]
This is the bungeecord bugtracker, not the spigot one.

**Comment by nikosgram13** [comment]
Yeah, report it here: https://hub.spigotmc.org/stash/projects/SPIGOT/repos/spigot/browse

**Comment by yawkat** [comment]
Oh, looks like you just forgot to adjust the text for bungeecord.

**Comment by nikosgram13** [comment]
@md-5 !

**Comment by HeroCC** [comment]
@yawkat, sorry, you are right, I just forgot to adjust it for spigot, I fixed it
